app/Console/Commands/Sync.php

```diff
@@ -31,21 +31,37 @@ class Sync extends Command
 	 */
 	public function handle(Exec $exec)
 	{
-		$this->info('Start syncing.');
 		$directory = $this->argument('dir');
 		$owner_id = (int) $this->option('owner_id'); // in case no ID provided -> import as root user
 		$album_id = (int) $this->option('album_id'); // in case no ID provided -> import to root folder
 
 		// Enable CLI formatting of status
 		$exec->statusCLIFormatting = true;
 		$exec->memCheck = false;
-		$exec->delete_imported = false; // we want to sync -> do not delete imported files
-		$exec->import_via_symlink = (Configs::get_value('import_via_symlink', '0') === '1');
-		$exec->skip_duplicates = true;
+		$exec->delete_imported = $this->option('delete_imported');
```

Comment:
I recommend an improvement here in order to avoid coding the same error message "--delete_imported [...] and [...] are conflicting" twice.

First check `if ($this->option('import_via_symlink') === '1' && $this->option('delete_imported') === '1')` even before you set `$exec->delete_imported` (i.e. before line 41) and if the condition holds, then bail out with an error message. Then you can be sure, that both options are not set at the same time in the following. This also avoids the nested `if` below.

Reply:
> Then you can be sure, that both options are not set at the same time in the following. 

If we only check for options with `if ($this->option('import_via_symlink') === '1' && $this->option('delete_imported'))`, how are we dealing with the Config setting `(Configs::get_value('import_via_symlink', '0') === '1') == true` and `$this->option('delete_imported')` ?
I've done it this way to make the sync backward compatible so that people using the config  `import_via_symlink === 1` don't have to add the `--import_via_symlink=1` flag.

Reply:
Could you push your forked branch `patch-1` from the repository `pchampio/Lychee ` as a new branch to `LycheeOrg/Lychee`? However, I would recommend to rename the branch to something else than `patch-1`.

Then I could checkout the branch myself and push a commit to show what I mean. I think you don't get the idea.

Reply:
I took the liberty to push a new commit. I even went beyond my original idea and I added the additional feature that the default values of the command line options are taken from the user configuration.

Reply:
@pchampio What do you think?

Reply:
This is much better!
Thanks, I was looking for a way to set the default values of the command with reference to the configuration, but didn't found how to do it.